src/StepperISR_due.cpp

```diff
@@ -1,401 +1,691 @@
+        q->timePWMInterruptEnabled = micros();                                 \
+        PWM_INTERFACE->PWM_IER1 = q->mapping->channelMask;                     \
+        port->PIO_CODR = samPin;                                               \
+        port->PIO_OER = samPin;                                                \
+        port->PIO_PER = samPin;                                                \
+        port->PIO_IDR = samPin;                                                \
+        PWM_INTERFACE->PWM_CH_NUM[channel].PWM_CPRDUPD = e->ticks;             \
+        return;                                                                \
+      }                                                                        \
+      /*That should be it...We should start the next command on the next       \
+        cycle */                                                               \
+      AddToTotalSteps(Q, e->steps);                                            \
+      PWM_INTERFACE->PWM_CH_NUM[channel].PWM_CPRDUPD = e->ticks;               \
+      if (e->toggle_dir) {                                                     \
+        e->toggle_dir = 0;                                                     \
+        *q->_dirPinPort ^= q->_dirPinMask;                                     \
+        delayMicroseconds(30);                                                 \
+      }                                                                        \
+      return;                                                                  \
+    }                                                                          \
+  }
+
+DUE_STEPPER_ISR(0)
+DUE_STEPPER_ISR(1)
+DUE_STEPPER_ISR(2)
+DUE_STEPPER_ISR(3)
+DUE_STEPPER_ISR(4)
+DUE_STEPPER_ISR(5)
+
+inline uint32_t pinToChannel(uint32_t pin) {
+  switch (pin) {
+    case 34:
+    case 67:
+    case 73:
+    case 35:
+      return 0;
+    case 17:
+    case 36:
+    case 72:
+    case 37:
+    case 42:
+      return 1;
+    case 38:
+    case 43:
+    case 63:
+    case 39:
+      return 2;
+    case 40:
+    case 64:
+    case 69:
+    case 41:
+      return 3;
+    case 9:
+      return 4;
+    case 8:
+    case 44:
+      return 5;
+    case 7:
+    case 45:
+      return 6;
+    case 6:
+      return 7;
+  }
+  return 0xFFFFFFFF;
+}
+
+void StepperQueue::init(uint8_t queue_num, uint8_t step_pin) {
+  _queue_num = queue_num;
+  mapping = &gChannelMap[queue_num];
+  _initVars();
+  _step_pin = step_pin;
+#if defined(KEEP_SCORE)
+  totalPulsesDetected[queue_num] = 0;
+  totalSteps[queue_num] = 0;
+#endif
+  gChannelMap[queue_num].pin = step_pin;
+  gChannelMap[queue_num].channel = pinToChannel(step_pin);
+  TimerChannel_Map[gChannelMap[queue_num].channel] = queue_num;
+  gChannelMap[queue_num].port = g_APinDescription[step_pin].pPort;
+  gChannelMap[queue_num].channelMask = 1 << gChannelMap[queue_num].channel;
+  static bool isPWMEnabled = false;
```

Comment:
In the above statements can replace `gChannelMap[queue_num].` with `mapping->`

Reply:
This code is run just once, and mapping is defined as const, since after the init function, it should never be written to.  I had the same idea as you, and had made the change last night.  It doesn't compile :)  the one that could be replaced is the the mask the channel will use, not the setting of the variable itself, but the read of channel used to shift.  Again, its in init, so I don't have emotions about it one way or the other, so if you want to cast it so it can be written to, or make the one possible change, feel free.  Doesn't seem like its worth it to me.  I'm totally willing to add a comment though :)

Reply:
True, but above the pointer is defined and not using it in the subsequent statements looks just inconsistent. But again, just a polishing thing.